hack/preload-images/preload_images.go

```diff
@@ -40,61 +42,106 @@ var (
 	dockerStorageDriver = "overlay2"
 	podmanStorageDriver = "overlay"
 	containerRuntimes   = []string{"docker", "containerd", "cri-o"}
-	k8sVersion          string
 	k8sVersions         []string
+	k8sVersion          = flag.String("kubernetes-version", "", "desired Kubernetes version, for example `v1.17.2`")
+	noUpload            = flag.Bool("no-upload", false, "Do not upload tarballs to GCS")
+	force               = flag.Bool("force", false, "Generate the preload tarball even if it's already exists")
+	limit               = flag.Int("limit", 0, "Limit the number of tarballs to generate")
 )
 
-func init() {
-	flag.StringVar(&k8sVersion, "kubernetes-version", "", "desired Kubernetes version, for example `v1.17.2`")
-	flag.Parse()
-	if k8sVersion != "" {
-		k8sVersions = append(k8sVersions, k8sVersion)
-	}
-	viper.Set("preload", "true")
+type preloadCfg struct {
+	k8sVer  string
+	runtime string
+}
+
+func (p preloadCfg) String() string {
+	return fmt.Sprintf("%q/%q", p.runtime, p.k8sVer)
 }
 
 func main() {
-	defer func() {
-		if err := deleteMinikube(); err != nil {
-			fmt.Printf("error cleaning up minikube: %v \n", err)
-		}
-	}()
+	flag.Parse()
+	viper.Set("preload", "true")
```

Comment:
if not used at all, lets remove if used, we can keep it

Reply:
we need this flag for `download.PreloadExists` method to work